lib/pf/vlan.pm

```diff
@@ -392,53 +401,42 @@ sub getNormalVlan {
-                $logger->trace("No unregdate found - computing it from access duration");
-                $value = access_duration($value);
-            }
-            else {
-                $value = &pf::authentication::match([@sources], $params, $Actions::SET_UNREG_DATE);
+        # Attributes has been computed in getNodeInfoForAutoReg
+        if ((isenabled($node_info->{'autoreg'}) && $autoreg) || !(isenabled($profile->dot1xRecomputeRoleFromPortal))) {
+            $logger->info("[$mac] Connection type is EAP. Getting role from node_info" );
+            $role = $node_info->{'category'};
+        } else {
+            my @sources = ($profile->getInternalSources, $profile->getExclusiveSources );
+            my $stripped_user = '';
+            $stripped_user = $stripped_user_name if(defined($stripped_user_name));
+            my $params = {
+                username => $user_name,
+                connection_type => connection_type_to_str($connection_type),
+                SSID => $ssid,
+                stripped_user_name => $stripped_user,
+            };
+            my $source;
+            $role = &pf::authentication::match([@sources], $params, $Actions::SET_ROLE, \$source);
+            # create a person entry for pid if it doesn't exist
+            if ( !pf::person::person_exist($user_name) ) {
+                $logger->info("creating person $user_name because it doesn't exist");
+                pf::person::person_add($user_name);
+                pf::lookup::person::lookup_person($user_name,$source);
+            } else {
+                $logger->debug("person $user_name already exists");
             }
-            if (defined $value) {
-                $value = pf::config::dynamic_unreg_date($value);
-                my %info = (
-                    'unregdate' => $value,
-                    'category' => $role,
-                    'autoreg' => 'yes',
-                    'pid' => $user_name,
-                );
-                if (defined $role) {
-                    %info = (%info, (category => $role));
-                }
-                # create a person entry for pid if it doesn't exist
-                if ( !pf::person::person_exist($user_name) ) {
-                    $logger->info("creating person $user_name because it doesn't exist");
-                    pf::person::person_add($user_name);
-                    pf::lookup::person::lookup_person($user_name,$source);
-                } else {
-                    $logger->debug("person $user_name already exists");
-                }
-                pf::person::person_modify($user_name,
-                    'source'  => $source,
-                    'portal'  => $profile->getName,
-                );
-                node_modify($mac,%info);
+            pf::person::person_modify($user_name,
+                'source'  => $source,
+                'portal'  => $profile->getName,
+            );
+            my %info = (
+                'autoreg' => 'no',
```

Comment:
We should not set autoreg no since we may be in here because we enabled dot1xRecomputeRoleFromPortal